pypit/calibrations.py

```diff
@@ -0,0 +1,512 @@
+""" Class for guiding calibration object generation in PYPIT
+"""
+from __future__ import absolute_import, division, print_function
+
+import inspect
+import numpy as np
+import os
+
+
+from abc import ABCMeta
+
+from pypit import msgs
+from pypit import ardebug as debugger
+from pypit import arpixels
+
+from pypit.core import arsort
+from pypit.core import armasters
+
+from pypit import arcimage
+from pypit import biasframe
+from pypit.spectrographs import bpmimage
+from pypit import flatfield
+from pypit import traceimage
+from pypit import traceslits
+from pypit import wavecalib
+from pypit import wavetilts
+from pypit import waveimage
+
+from pypit import ardebug as debugger
+
+from pypit.spectrographs import io
+
+# For out of PYPIT running
+if msgs._debug is None:
+    debug = debugger.init()
+    debug['develop'] = True
+    msgs.reset(debug=debug, verbosity=2)
+
+# Does not need to be global, but I prefer it
+frametype = 'bias'
+
+# Place these here or elsewhere?
+#  Wherever they be, they need to be defined, described, etc.
+#  These are settings beyond those in the Parent class (ProcessImages)
+additional_default_settings = {frametype: {'useframe': 'none'}}
+
+
+class Calibrations(object):
+    """
+    This class is primarily designed to guide the generation of calibration images
+    and objects in PYPIT
+
+    Parameters
+    ----------
+
+    Attributes
+    ----------
+
+    Inherited Attributes
+    --------------------
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, fitstbl, save_masters=True, write_qa=True):
+
+        # Parameters unique to this Object
+        self.fitstbl = fitstbl
+        self.save_masters = save_masters
+        self.write_qa = write_qa
+
+        # Set spectrograph from FITS table
+        self.spectrograph = self.fitstbl['instrume'][0]
```

Comment:
Thanks you @profxj, I see you are setting instruments from headers now. Much appreciated.